### PR TITLE
Backfill some more missing javadoc (api module)

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Chipping away at the yellow deluge from the maven build.

### Additional Context

_Why are you making this pull request?_

### CODE REVIEW

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java [hunk 1]
```diff
@@ -16,24 +16,30 @@
...
-    private final Map<String, InstanceBuilder<? extends BaseConfig, L>> shortNameToInstanceBuilder;
+    private final Map<String, InstanceBuilder<? extends BaseConfig, T>> shortNameToInstanceBuilder;
 
-    public BaseContributor(BaseContributorBuilder<L> builder) {
+    /**
+     * Constructs the contributor using the supplied {@code builder}.
+     * @param builder builder
+     */
+    public BaseContributor(BaseContributorBuilder<T> builder) {
         shortNameToInstanceBuilder = builder.build();
```
Comment (SamBarker) [inline api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java:31]: I know its not part of this PR, but `shortNameToInstanceBuilder = builder.build();` surprised me. The builder is building a map not a thing?

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java [hunk 2]
```diff
@@ -16,24 +16,30 @@
...
  */
-public abstract class BaseContributor<L> implements Contributor<L> {
+public abstract class BaseContributor<T> implements Contributor<T> {
 
-    private final Map<String, InstanceBuilder<? extends BaseConfig, L>> shortNameToInstanceBuilder;
+    private final Map<String, InstanceBuilder<? extends BaseConfig, T>> shortNameToInstanceBuilder;
 
-    public BaseContributor(BaseContributorBuilder<L> builder) {
+    /**
+     * Constructs the contributor using the supplied {@code builder}.
```
Comment (SamBarker) [inline api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java:27]:
```suggestion
     * Constructs and configures the contributor using the supplied {@code builder}.
```

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java [hunk 3]
```diff
@@ -62,13 +68,27 @@ else if (configClass.isAssignableFrom(config.getClass())) {
...
+     */
     public static class BaseContributorBuilder<L> {
 
         private BaseContributorBuilder() {
         }
 
         private final Map<String, InstanceBuilder<?, L>> shortNameToInstanceBuilder = new HashMap<>();
 
+        /**
+         * Registers a service instance constructor.
```
Comment (SamBarker) [inline api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java:84]: I think technically it registers a factory function.

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java [hunk 4]
```diff
@@ -11,10 +11,26 @@
...
+    /**
+     * Gets the concrete type of the configuration required by this service instance.
+     * @param shortName service short name
+     *
+     * @return class of a concrete type or null, if this service instance does not accept configuration.
+     */
     Class<? extends BaseConfig> getConfigType(String shortName);
 
+    /**
+     * Creates a instance of the service.
```
Comment (SamBarker) [inline api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java:28]: For my money `creates a instance` implies a different contract to the method name `getInstance`. The method name implies caching/singletons are supported however `creates a instance` implies that every invocation gets a new instance. 

I'm not sure which contract we should prefer or mean to imply but I think we should be consistent between the javadoc and method names.

Also while I'm here in the grammar pedant category shouldn't it be 
```suggestion
     * Creates an instance of the service.
```

**Review by SamBarker — APPROVED** [review]
Thanks for picking this up @k-wall

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java (hunk 1 above)
Comment (k-wall) [reply api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java:31]: Agreed, the naming is off.  `BaseContributorBuilder`=> `ContributorRegistryBuilder`.  ContributorRegistryBuilder should build a `ContributorRegistry` which might be composed of a Map.  Obviously not part of this PR.  @robobario

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java (hunk 4 above)
Comment (SamBarker) [reply api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java]: While https://github.com/kroxylicious/kroxylicious/commit/e47f523d8fbeadfcfcc910989ef77de850ed6ffd fixes the nit. It doesn't address the wider point I was making about the consistency between the doc and the method name.

Comment (k-wall) [reply api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java]: Good point.  I've opted to rename the method.  I think the simpler semantics are good enough for now.  If we took the alternative path, we'd need to think about the immutability of the `getInstance()'s` parameters and potentially lifecycling of the service instance.

**Comment by k-wall** [comment]
@robobario if you are happy, please go ahead and merge.

File: api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java (hunk 4 above)
Comment (robobario) [reply api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java]: Yep, sounds good.
